istioctl/cmd/workload.go

```diff
@@ -0,0 +1,399 @@
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			wg := &clientv1alpha3.WorkloadGroup{}
+			kubeClient, err := kube.NewExtendedClient(kube.BuildClientCmd(kubeconfig, configContext), "")
+			if err != nil {
+				return err
+			}
+			if filename != "" {
+				if err := readWorkloadGroup(filename, wg); err != nil {
+					return err
+				}
+			} else {
+				// TODO: figure out why workloadgroups isn't a recognized resource
+				wg, err = networkingclientv1alpha3.New(kubeClient.REST()).WorkloadGroups(namespace).Get(context.Background(), name, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+			}
+
+			if err = createConfig(kubeClient, wg, clusterID, revision, outputName); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	configureCmd.PersistentFlags().StringVarP(&filename, "file", "f", "", "filename of the WorkloadGroup artifact")
+	configureCmd.PersistentFlags().StringVar(&name, "name", "", "The name of the workload group")
+	configureCmd.PersistentFlags().StringVarP(&namespace, "namespace", "n", "", "The namespace that the workload instances belongs to")
+	configureCmd.PersistentFlags().StringVarP(&outputName, "output", "o", "", "Name of the tarball to be created")
+	configureCmd.PersistentFlags().StringVar(&clusterID, "clusterID", "", "The ID used to identify the cluster")
+	configureCmd.PersistentFlags().Int64Var(&tokenDuration, "tokenDuration", 3600, "The token duration in seconds (default: 1 hour)")
+	configureCmd.PersistentFlags().StringVar(&revision, "revision", "", "control plane revision (experimental)")
+	return configureCmd
+}
+
+func readWorkloadGroup(filename string, wg *clientv1alpha3.WorkloadGroup) error {
+	f, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	if err = yaml.Unmarshal(f, wg); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Creates all the relevant config for the given workload group and cluster
+func createConfig(kubeClient kube.ExtendedClient, wg *clientv1alpha3.WorkloadGroup, clusterID, revision, outputName string) error {
+	temp, err := ioutil.TempDir("./", outputName)
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(temp)
+
+	if err = createClusterEnv(wg, temp); err != nil {
+		return err
+	}
+	if err = createCertsTokens(kubeClient, wg, temp); err != nil {
+		return err
+	}
+	if err = createMeshConfig(kubeClient, wg, clusterID, revision, temp); err != nil {
+		return err
+	}
+	if err = Tar(temp, outputName); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Write cluster.env into the given directory
+func createClusterEnv(wg *clientv1alpha3.WorkloadGroup, dir string) error {
+	we := wg.Spec.Template.Spec
+	for _, v := range we.Ports {
+		ports = append(ports, fmt.Sprint(v))
+	}
+	// default attributes and service name, namespace, ports, service account, service CIDR
+	clusterEnv := map[string]string{
+		"ISTIO_CP_AUTH":       "MUTUAL_TLS",
+		"ISTIO_INBOUND_PORTS": strings.Join(ports, ","),
```

Comment:
I am pretty sure this will cause issues. The port field here will not be set in majority of cases. We actually probably want to capture all inbound traffic and respect the inbound port annotation (see https://istio.io/latest/docs/reference/config/annotations/)